lld/MachO/Driver.cpp

```diff
@@ -1337,10 +1338,25 @@ static void codegenDataGenerate() {
       // Merge it to the global hash tree.
       globalOutlineRecord.merge(localOutlineRecord);
     }
+    if (isec->getSegName() == segment_names::data &&
+        isec->getName() == section_names::functionmap) {
+      // Read stable functions from each section.
+      StableFunctionMapRecord localMergeRecord;
+      auto *data = isec->data.data();
```

Comment:
any reason `data` here is a separate variable ?

Reply:
Yes because of `void deserialize(const unsigned char *&Ptr)` which modifies/advances the pointer. I added a comment above.